packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -670,6 +670,10 @@ abstract class DatabaseInstanceSource extends DatabaseInstanceNew implements IDa
   constructor(scope: Construct, id: string, props: DatabaseInstanceSourceProps) {
     super(scope, id, props);
 
+    if (/aurora/.test(props.engine.name)) {
```

Comment:
I'm not sure I love this test...

How about a method on `Engine`, something like `get allowedOutsideCluster(): boolean` (name TBD of course)? Then, this code could be:

```ts
    if (!props.engine.allowedOutsideCluster) {
      throw new Error(`${props.engine.name} instances can only be created inside a cluster. Please use the DatabaseCluster class instead`);
    }
```

What do you think?
```

Reply:
Yes, it's cleaner.

Reply:
Actually just a `public readonly isDatabaseInstanceEngine = true;` in `DatabaseInstanceEngine` is enough no? Because then the typing system will not allow to use a `DatabaseInstanceCluster` for a `DatabaseInstance` and we don't need to check anymore?

Reply:
Yep. This works:

```ts
export class DatabaseInstanceEngine extends DatabaseClusterEngine {
  // ...

  /** To make it a compile-time error to pass a DatabaseClusterEngine where a DatabaseInstanceEngine is expected. */
  public readonly isDatabaseInstanceEngine = true;
}
```

Having it be a compile-time error is even better than adding runtime validation.